meilisearch/src/middleware.rs

```diff
@@ -47,19 +49,29 @@ where
 
     fn call(&self, req: ServiceRequest) -> Self::Future {
         let mut histogram_timer: Option<HistogramTimer> = None;
-        let request_path = req.path();
-        let is_registered_resource = req.resource_map().has_resource(request_path);
-        if is_registered_resource {
-            let request_method = req.method().to_string();
-            histogram_timer = Some(
-                crate::metrics::MEILISEARCH_HTTP_RESPONSE_TIME_SECONDS
+        let data = req.app_data::<Data<IndexScheduler>>();
+        let metrics_enabled = data
+            .ok_or("Could not get index scheduler")
+            .and_then(|index_scheduler| {
+                index_scheduler.features().map_err(|_| "Could not get features")
+            })
+            .and_then(|features| features.check_metrics().map_err(|_| "Metrics not enabled"));
+
+        if let Ok(()) = metrics_enabled {
```

Comment:
It would indeed be better to return an error. It would be much better than the current strings not being displayed in case of error here. Would it be possible to return a 500 with the error message? Even an ugly internal one?

Reply:
Will update! I was worried about this middleware failure taking down the ability to handle any requests so I didn't want to return the error off the bat. I should've left a comment asking about expected behavior there. I can see that you wouldn't want metrics to disappear silently